custom_components/ha_washdata/profile_store.py

```
@@ -5623,17 +6044,335 @@ async def async_import_data(self, payload: dict[str, Any]) -> dict[str, Any]:
+
+        # Replace-mode anti-data-loss guard: never wipe a destination unless the file has a
+        # SELECTED set that will actually refill it. Guarding on the post-filter sets (not raw
+        # src_*) means a stale/malformed id selection that matches nothing aborts here instead
+        # of emptying the local list; for reference-bound cycles we further require at least
+        # one usable trace so an all-degenerate selection can't wipe-then-add-nothing. Profiles
+        # are never wiped (per-name overwrite/copy), so a profiles-only replace needs no guard.
+        if mode == "replace":
+            if "reference_cycles" in cats and not _any_usable(selected_refs):
+                raise ValueError(
+                    "Import payload has no usable reference cycles — aborting to prevent data loss"
+                )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)